Modules/posixmodule.c

```diff
@@ -4243,7 +4245,8 @@ _posix_listdir(path_t *path, PyObject *list)
     }
 
     if (dirp == NULL) {
-        list = path_error(path);
+        path_error(path);
+        list = NULL;
```

Comment:
This list **parameter** is like **really** surprising. Would you mind to take the opportunity to convert it to a regular local variable (and maybe always initialize it to NULL at the beginning)?

It seems like in the case, the caller could pass their own list.

I have a personal preference to code like this for the code below:

```
    list = PyList_New(0);
    if (list == NULL) {
        goto exit;
    }
```